### ISSUE TITLE
Plugin not found

### ISSUE THREAD

**Issue by joezappie** [IhToN/verdaccio-azure-ad-login#1]
I'm getting a "Plugin not found" error with this module. I read that if a plugin is not valid that it shows this message as well. Does this need to get updated to the latest Verdaccio?

Installing another plugin verdaccio-activedirectory that loads correctly and both are getting installed to the same location.

**Comment by IhToN** [comment]
Hello jrj2211, I've not been able to test it on the latest verdaccio version but as far as I know it's working on 4.6.2.

I'll try to give it a look.

**Comment by IhToN** [comment]
Hi @jrj2211. I've just tested it today over Verdaccio 4.11.0 and it's working as intended. Can you provide a log file or, if you don't mind, a censored version of the configuration you are trying to apply?

**Comment by joezappie** [comment]
Hi @IhToN, thanks for taking a look. I've still been unable to get mine to load. Here some more info on my system:

I'm running Linux Debian Stretch 9 with sudo permissions. I've installed the azure-ad plugin globally with
`try npm install verdaccio-azure-ad-login -g`

Looking in the global folder I can see that it is there, along with the other activedirectory plugin that verdaccio loads:
```
npm list -g --depth=0
/usr/local/lib
├── nodemon@2.0.6
├── npm@6.14.8
├── peer@0.5.3
├── verdaccio@4.11.0
├── verdaccio-activedirectory@1.0.2
└── verdaccio-azure-ad-login@1.0.3
```

When running "verdaccio" in the command line, here is the output log (config changed to log debug):
```
 warn --- config file  - /root/.config/verdaccio/config.yaml
 warn --- Verdaccio started
 debug--- [local-storage/_sync]: init sync database
 debug--- [local-storage/_sync]: folder /root/.config/verdaccio/storage created succeed
 debug--- [local-storage/_sync/writeFileSync]: sync write succeed
 debug--- [local-storage/_sync]: init sync database
 debug--- [local-storage/_sync]: folder /root/.config/verdaccio/storage created succeed
 debug--- [local-storage/_sync/writeFileSync]: sync write succeed
 warn --- Plugin successfully loaded: verdaccio-htpasswd
 error--- plugin not found. try npm install verdaccio-azure-ad-login
(node:18219) UnhandledPromiseRejectionWarning: Error:
        verdaccio-azure-ad-login plugin not found. try "npm install verdaccio-azure-ad-login"
    at /usr/local/lib/node_modules/verdaccio/build/lib/plugin-loader.js:110:13
    at Array.map (<anonymous>)
    at loadPlugin (/usr/local/lib/node_modules/verdaccio/build/lib/plugin-loader.js:62:37)
    at Auth._loadPlugin (/usr/local/lib/node_modules/verdaccio/build/lib/auth.js:54:38)
    at new Auth (/usr/local/lib/node_modules/verdaccio/build/lib/auth.js:44:25)
    at defineAPI (/usr/local/lib/node_modules/verdaccio/build/api/index.js:43:16)
    at _default (/usr/local/lib/node_modules/verdaccio/build/api/index.js:127:10)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:18219) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:18219) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Here is my the important parts of the config file slightly censored (I'll regenerate these anyways)
```
storage: ./storage
plugins: ./plugins

web:
  title: Verdaccio
  darkMode: true

# translate your registry, api i18n not available yet
# i18n:
# list of the available translations https://github.com/verdaccio/ui/tree/maste$
#   web: en-US

auth:
  htpasswd:
    file: /root/.config/verdaccio/.htpasswd
  azure-ad-login:
    # REQUIRED, Azure application tenant
    tenant: "XXXXXXXX-XXXX-XXXX-XXXX-6a0ebdbcb656"
    # REQUIRED, Azure client_id
    client_id: "XXXXXXXX-XXXX-XXXX-XXXX-51cd0c5101a5"
    # REQUIRED, Azure application client_secret
    client_secret: "XXXXXXXXly5K_n6bKrA_7Rf4bI9o_s~e~2"
```

If it is my config file thats causing it to not load properly, I am a little confused on what the tenant field should be. It should be an ID? And client_id should be the other ID listed in the app registration?

![image](https://user-images.githubusercontent.com/6005836/109042586-3b25e800-7685-11eb-9353-1eff586b1308.png)

Also I was confused about the client_secret - you get both a value and ID when you generate a secret in azure. It should just be the value and the ID is unused?

![image](https://user-images.githubusercontent.com/6005836/109042481-1a5d9280-7685-11eb-90d3-4ef4a43f1098.png)

Thanks for the help!

**Comment by IhToN** [comment]
Hi @jrj2211, your config is correct. Regarding your doubt about the client_secret the only thing that the plugin is using is the "ID", the value is ignored as it is not needed to get the information.

Also, I found the issue, it looks like that the tarball of the version 1.0.3 was empty, so only the "package.json" file was being downloaded. I've released a new 1.0.4 version that should fix the issue.

Please tell me once you try and, if everything works as expected, I'll close the issue.

Sorry for the inconvenience and thanks for trusting this plugin :)

**Comment by joezappie** [comment]
Woooo! That fixed it. Thank you for the quick fix and providing instructions on the config file!

`warn --- Plugin successfully loaded: verdaccio-azure-ad-login`
